lib/rasn1/types/base.rb

```diff
@@ -360,6 +360,7 @@ def build
           if explicit?
             v = explicit_type
             v.value = @value
+            v.bit_length = @bit_length if @bit_length
```

Comment:
I don't think this is the complete fix; It looks like it might be better to use something like `initialize_copy` instead of `explicit_type` here - but this follows the existing pattern and I wanted to put up a PR for feedback first :+1:

Reply:
Hi @adfoster-r7 Thank you for your interest in rasn1.

I think the best fix is to redefine `explicit_type` in `BitString` class, as this is done in `Integer`.

The `Base` class should have no knowledge of BitString internals.